crates/walrus-service/src/client/daemon/routes.rs

```diff
@@ -123,13 +113,48 @@ pub(super) async fn get_blob<T: WalrusReadClient>(
     }
 }
 
-fn populate_response_headers(
+fn populate_response_headers_from_request(
+    request_headers: &HeaderMap,
+    etag: &str,
+    headers: &mut HeaderMap,
+) {
+    // Prevent the browser from trying to guess the MIME type to avoid dangerous inferences.
+    headers.insert(X_CONTENT_TYPE_OPTIONS, HeaderValue::from_static("nosniff"));
+    // Insert headers that help caches distribute Walrus blobs.
+    //
+    // Cache for 1 day, and allow refreshig on the client side. Refreshes use the ETag to
+    // check if the content has changed. This allows invalidated blobs to be removed from
+    // caches. `stale-while-revalidate` allows stale content to be served for 1 hour while
+    // the browser tries to validate it (async revalidation).
+    headers.insert(
+        CACHE_CONTROL,
+        HeaderValue::from_static("public, max-age=86400, stale-while-revalidate=3600"),
+    );
+    // The `ETag` is the blob ID itself.
+    headers.insert(
+        ETAG,
+        HeaderValue::from_str(etag)
+            .expect("the blob ID string only contains visible ASCII characters"),
+    );
+    // Mirror the content type.
+    if let Some(content_type) = request_headers.get(CONTENT_TYPE) {
+        tracing::debug!(?content_type, "mirroring the request's content type");
+        headers.insert(CONTENT_TYPE, content_type.clone());
+    } // Cache for 1 day, and allow refreshig on the client side. Refreshes use the ETag to
+}
+
+fn populate_response_headers_from_attributes(
     headers: &mut HeaderMap,
     attribute: &BlobAttribute,
-    allowed_headers: &HashSet<String>,
+    allowed_headers: Option<&HashSet<String>>,
 ) {
     for (key, value) in attribute.iter() {
-        if allowed_headers.contains(key) {
+        tracing::info!(
+            attribute_key = key,
+            attribute_value = value,
+            "quilt blob attribute"
```

Comment:
Nit: maybe just "blob attribute", since this is used in both cases? Also should this log be in DEBUG level?